Docs/source/How_To/coding-style.rst

```diff
@@ -0,0 +1,271 @@
+also be separated by snakecase and describe the function's purpose. If there are multiple words in
+the third section they should be written in CamelCase. Ex
+
+.. code-block:: c
+
+    Layer_Library_Function();
+    BSP_Contactor_GetState();
+
+Descriptions for public functions should be included above the function definition in the source
+file and above the declaration in the header file.
+
+=================
+Private Functions
+=================
+
+Code duplication should be avoided by using private functions instead. Their format is the same as public functions but 
+do not need to have the layer section in their name. Their description should be included in the source file, not the header file.
+
+.. code-block:: c
+
+    /**
+     * NOTE: Include anything important someone else will need to know
+     * @brief Give Description of Function
+     * @param variable Describe input parameters
+     * @return What function returns
+     */
+    static void ADC_InitDMA(void);
+
+Variables
+=========
+
+================
+Local Variables
+================
+
+If variables have the same type, declare them on the same line if possible. 
+Meaningful names for variables that aren't used for iterations in a loop should be written in camelCase.
+Unnecessary variables should be avoided. i.e
+
+.. code-block:: c
+    :linenos:
+
+    //NO BUENO
+    int x = function();
+    if (x) return;
+    //YES BUENO
+    if (function()) return;
+
+If the functionality of the code is affected by this (calling the same function multiple times), then
+it is fine to have a variable for the return value.
+
+================
+Global Variables
+================
+
+Variable names should be meaningful and written in CamelCase. Efforts must be made to prevent global variables from
+being used in multiple libraries. All global variables must be static variables unless the logic requires global variables
+being shared. All global variables should have comments describing their purpose (what they hold). Ex.
+
+.. code-block:: c
+    :linenos:
+
+    static OS_MUTEX VoltageMutex;
+    static uint16_t VoltageVal[NUM_BATTERY_MODULES]; //Voltage values gathered
+
+Miscellaneous
+=============
+
+============
+Punctuation
+============
+
+**Curly Braces:** Opening braces should be on the same line as the condition or function that contains them. Curly braces should always 
+be used in conditionals even if it is one line
+**Parentheses:** Should be used for clarifying the order of precedence.
+**Indentation:** Indentation should be used in all conditionals, iteratives, and functions. Indentation should be 4 spaces (NOT TABS).
+**Line Endings:** All End Of Line sequences should be LF(line feed). This prevents files from looking like they are changed when
+nothing was changed at all.
+
+===============
+Common Practice
+===============
+
+**Pointers:** Members to pointers should be accessed through ``p->member`` operator instead of ``(p).member``.
```

Comment:
```suggestion
**Pointers:** Members to pointers should be accessed through ``p->member`` operator instead of ``(*p).member``.
```
I assume this was a typo